sunscreen_tfhe/src/entities/bootstrap_key.rs

```diff
@@ -69,13 +69,26 @@ impl<S: TorusOps> BootstrapKeyRef<S> {
     pub fn fft(
         &self,
         result: &mut BootstrapKeyFftRef<Complex<f64>>,
-        params: &GlweDef,
+        lwe: &LweDef,
+        glwe: &GlweDef,
         radix: &RadixDecomposition,
     ) {
-        for (s, r) in self.rows(params, radix).zip(result.rows_mut(params, radix)) {
-            s.fft(r, params, radix);
+        self.assert_valid(lwe, glwe, radix);
+        result.assert_valid(lwe, glwe, radix);
+
+        for (s, r) in self.rows(glwe, radix).zip(result.rows_mut(glwe, radix)) {
+            s.fft(r, glwe, radix);
         }
     }
+
+    #[inline(always)]
+    /// Asserts that this entity is valid under the passed parameters.
+    pub fn assert_valid(&self, lwe: &LweDef, glwe: &GlweDef, radix: &RadixDecomposition) {
```

Comment:
I believe some of our other `assert_valid` functions are `pub(crate)` instead of `pub`. Seems like we should convert them all to `pub` I'm guessing.